tests/test_crawl.py

```diff
@@ -226,3 +226,13 @@ def cb(response):
         s = dict(est[0])
         self.assertEqual(s['engine.spider.name'], crawler.spider.name)
         self.assertEqual(s['len(engine.scraper.slot.active)'], 1)
+
+    @defer.inlineCallbacks
+    def test_graceful_crawl_error_handling(self):
+        crawler = get_crawler(SimpleSpider)
+        class TestError(Exception):
+            pass
+        with mock.patch('scrapy.crawler.ExecutionEngine.open_spider') as mock_os:
```

Comment:
Could you please add a comment how can this error happen? Is it an error raised in some user component? I have troubles reading tests which use mocks, sorry :)